src/databricks/labs/ucx/assessment/crawlers.py

```diff
@@ -78,6 +129,24 @@ def _assess_clusters(self, all_clusters):
                 for value in cluster.spark_conf.values():
                     if "dbfs:/mnt" in value or "/dbfs/mnt" in value:
                         failures.append(f"using DBFS mount in configuration: {value}")
+
+                # Checking if Azure cluster config is present in spark config
+                for conf in _AZURE_SP_CLUSTER_CONF:
+                    if re.search(conf, str(cluster.spark_conf)):
+                        failures.append("Uses azure service principal credentials in cluster config.")
+
+            # Checking if Azure cluster config is present in cluster policies
+            if cluster.policy_id:
+                cluster_policy_definition = self._ws.cluster_policies.get(cluster.policy_id).definition
```

Comment:
You make two API different calls for the same id. This is a bug, as it's going to produce incorrect results when policy is edited between the calls